lldb/source/Plugins/Language/Swift/SwiftLanguage.cpp

```diff
@@ -794,33 +795,41 @@ ExtractSwiftTypeFromCxxInteropType(CompilerType type, TypeSystemSwift &ts,
     if (!swift::Demangle::isMangledName(swift_name))
       break;
 
-    swift_type = ts.GetTypeFromMangledTypename(ConstString(swift_name));
+    ts = get_swift_type_system();
+    if (!ts)
+      return {};
+
+    swift_type = ts->GetTypeFromMangledTypename(ConstString(swift_name));
```

Comment:
Do you think you could change this loop to fid the `swift_name` instead, and get the type system after the loop, only if you found a name? This way you don't need to have `TypeSystemSwift *ts = nullptr;` and setting it inside the statement.

Reply:
Sure.